lib/Core/Executor.cpp

```diff
@@ -867,14 +867,62 @@ bool Executor::branchingPermitted(const ExecutionState &state) const {
   return true;
 }
 
+/// Helper method to get an Assignment for a state.  This method is called
+/// when we have reached the maximum number of forks allotted for a particular
+/// exploration.
+Assignment * generateAssignmentForDanglingState(std::map<ExecutionState*, Assignment*> &maxForksMap,
+                                                TimingSolver * solver, ExecutionState &current){
+  Assignment *a = maxForksMap[&current];
+  if (!a && current.symbolics.size() != 0){
+    std::vector< std::vector<unsigned char> > values;
+    std::vector<const Array*> objects;
+    for (unsigned i = 0; i != current.symbolics.size(); ++i){
+      objects.push_back(current.symbolics[i].second);
+    }
+    ConstraintSet extendedConstraints(current.constraints);
+    bool success = solver->getInitialValues(extendedConstraints, objects, values, current.queryMetaData);
+
+    a = new Assignment(objects, values);
+    maxForksMap[&current] = a;
+  }
+  return a;
+}
+
+bool reachMaxForks(){
```

Comment:
The way this is written is quite verbose.  Just replace it with`return (MaxForks!=~0u && stats::forks >= MaxForks)`

Reply:
I would also call this `reachedMaxForks` instead of `reachMaxForks`